tests/acceptance/tests/Account/CustomerCredentials.spec.ts

```diff
@@ -0,0 +1,69 @@
+import { test } from '@fixtures/AcceptanceTest';
+
+test('As a customer, I must be able to change email via account.', { tag: '@Account' }, async ({
+    ShopCustomer,
+    StorefrontAccountLogin,
+    StorefrontAccount,
+    IdProvider,
+    Register,
+    StorefrontAccountProfile,
+
+}) => {
+
+    const customer = { email: IdProvider.getIdPair().uuid + '@test.com' , password: IdProvider.getIdPair().uuid };
+    const changedEmail = { email: IdProvider.getIdPair().uuid + '@test.com' };
+
+    await ShopCustomer.goesTo(StorefrontAccountLogin.url());
+    await ShopCustomer.attemptsTo(Register(customer));
+    await ShopCustomer.expects(StorefrontAccount.page.getByText(customer.email, { exact: true })).toBeVisible();
+    await ShopCustomer.goesTo(StorefrontAccountProfile.url());
+    await StorefrontAccountProfile.changeEmailButton.click();
+    await ShopCustomer.expects(StorefrontAccountProfile.emailAddressInput).toBeVisible();
+    await StorefrontAccountProfile.emailAddressInput.fill(changedEmail.email);
+    await StorefrontAccountProfile.emailAddressConfirmInput.fill(changedEmail.email);
+    await StorefrontAccountProfile.emailConfirmPasswordInput.fill(customer.password);
+    await StorefrontAccountProfile.saveEmailAddressButton.click();
+    await ShopCustomer.expects(StorefrontAccountProfile.emailUpdateMessage).toBeVisible();
+    await ShopCustomer.expects(StorefrontAccountProfile.loginDataEmailAddress).toContainText(changedEmail.email);
+
+    await StorefrontAccountLogin.logoutLink.click();
+    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
+    await StorefrontAccountLogin.emailInput.fill(changedEmail.email);
+    await StorefrontAccountLogin.passwordInput.fill(customer.password);
+    await StorefrontAccountLogin.loginButton.click();
+    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
```

Comment:
To enhance the clarity and structure of your test script, you can break it down into logical steps using test.step. This approach helps others quickly understand the sequence of actions and the expected outcomes.

```suggestion
    await test.step('Navigate to login page and register a new customer', async () => {
    await ShopCustomer.goesTo(StorefrontAccountLogin.url());
    await ShopCustomer.attemptsTo(Register(customer));
    await ShopCustomer.expects(StorefrontAccount.page.getByText(customer.email, { exact: true })).toBeVisible();
});

await test.step('Change the registered customer\'s email address', async () => {
    await ShopCustomer.goesTo(StorefrontAccountProfile.url());
    await StorefrontAccountProfile.changeEmailButton.click();
    await ShopCustomer.expects(StorefrontAccountProfile.emailAddressInput).toBeVisible();
    await StorefrontAccountProfile.emailAddressInput.fill(changedEmail.email);
    await StorefrontAccountProfile.emailAddressConfirmInput.fill(changedEmail.email);
    await StorefrontAccountProfile.emailConfirmPasswordInput.fill(customer.password);
    await StorefrontAccountProfile.saveEmailAddressButton.click();
    await ShopCustomer.expects(StorefrontAccountProfile.emailUpdateMessage).toBeVisible();
    await ShopCustomer.expects(StorefrontAccountProfile.loginDataEmailAddress).toContainText(changedEmail.email);
});

await test.step('Log out and verify the logout success message', async () => {
    await StorefrontAccountLogin.logoutLink.click();
    await ShopCustomer.expects(StorefrontAccountLogin.successAlert).toBeVisible();
});

await test.step('Log in with the new email address', async () => {
    await StorefrontAccountLogin.emailInput.fill(changedEmail.email);
    await StorefrontAccountLogin.passwordInput.fill(customer.password);
    await StorefrontAccountLogin.loginButton.click();
    await ShopCustomer.expects(StorefrontAccount.personalDataCardTitle).toBeVisible();
});
```

Reply:
Added some steps 👍 Could you please check again